test/WebJobs.Script.Tests.Integration/TestFunctionHost.cs

```diff
@@ -53,18 +53,19 @@ public TestFunctionHost(string appRoot, Action<IHostBuilder> configureJobHost)
                      services.Replace(ServiceDescriptor.Singleton<IServiceProviderFactory<IServiceCollection>>(new WebHostServiceProviderFactory()));
 
                      services.Replace(new ServiceDescriptor(typeof(IOptions<ScriptApplicationHostOptions>), new OptionsWrapper<ScriptApplicationHostOptions>(_hostOptions)));
-                     services.Replace(new ServiceDescriptor(typeof(ISecretManager), new TestSecretManager()));
                  })
-                 .AddScriptHostBuilder(b =>
+                 .AddScriptHostBuilder(webJobsBuilder =>
                  {
-                     b.ConfigureLogging(l =>
-                     {
-                         l.AddProvider(_loggerProvider);
-                         l.AddFilter<TestLoggerProvider>(_ => true);
-                     })
-                     .AddAzureStorage();
-
-                     configureJobHost?.Invoke(b);
+                     var loggingBuilder = new LoggingBuilder(webJobsBuilder.Services);
```

Comment:
This feels a little clunky. Are you trying to restrict ScriptHostBuilder changes to only happen against the IWebJobsBuilder? Should it expose the IWebHostBuilder as well?

Reply:
The `IWebJobsBuilder` is already exposed (that's what you own and create above). We'd need to pass the `IHostBuilder` instance if we want the extension methods it provides. For most cases, this won't be an issue, but I'll send an update to make this cleaner in a separate PR